jdbc-dbcp/src/main/java/io/micronaut/configuration/jdbc/dbcp/DatasourceConfiguration.java

```diff
@@ -176,4 +178,22 @@ public void setDataSourceProperties(@MapFormat(transformation = MapFormat.MapTra
     public String getConfiguredValidationQuery() {
         return super.getValidationQuery();
     }
+
+    @Override
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * Sets an indicator telling whether data source is enabled.
+     * @param enabled an indicator telling whether data source is enabled
+     */
+    public void setEnabled(boolean enabled) {
```

Comment:
can we make this package private or annotate with `@Internal`? Since it always throws probably not ideal

Reply:
I marked class with `@Internal` and made non `public`. Tested with micronaut-data and examples used to reproduce the original issues, seems to be working ok.

Reply:
class can't be made non-public it would be a breaking change

Reply:
Yes, reverted it and updated method only so it works as needed.